google_guest_agent/network/manager/manager.go

```diff
@@ -244,6 +230,32 @@ func SetupInterfaces(ctx context.Context, config *cfg.Sections, mds *metadata.De
 	return nil
 }
 
+// Remove only primary nics left over configs.
+func rollbackLeftoverConfigs(ctx context.Context, config *cfg.Sections, mds *metadata.Descriptor) error {
+	if config.NetworkInterfaces.ManagePrimaryNIC {
+		return nil
```

Comment:
shouldn't this be restoring netplan configs regardless of we're managing primary nic or not to match current behavior?

Reply:
Yep, restoring netplan should happen regardless and nic rollback should be conditioned to both netplan restore and being managing primary nic. I'll adjust.

Reply:
done.